javascript-typing-tutor/script.js

```diff
@@ -0,0 +1,56 @@
+var value = 0;
+var attempts = 0;
+var $text = document.querySelectorAll('span');
+var $score = document.querySelector('p');
+var $retry = document.querySelector('button');
+var $modal = document.querySelector('.modal');
+var $modalText = document.querySelector('.modal-text');
+var $background = document.querySelector('body');
+var $opacity = document.querySelector('.text');
+var $retry2 = document.querySelector('.retry2');
+
+var test = true;
+document.addEventListener('keydown', function (e) {
+  if (e.key === $text[value].innerText) {
+    $text[value].className = 'green-text';
+    attempts++;
+    value++;
+    $score.textContent = 'Score: ' + value + '/27';
+    if (value === 27) {
+      $modal.className = 'modal';
+      $background.className = 'background-color-off';
+      $opacity.className = 'opacity text';
+      $modalText.textContent = Math.round(value / attempts * 100) + '% with ' + (attempts - value) + ' errors.';
+      $retry2.addEventListener('click', function () {
+        if (test === true) {
+          location.reload();
+        }
+      });
```

Comment:
Oh nooooooo 😱  

`location.reload` is a forbidden tool. It completely reloads the page, which is an **_extremely_** intensive process for the browser. Instead, you should reset all your elements to the state they would be in when the page loads using javascript, and reset the value of any variables in JavaScript to what their initial value would be as well.